File: 1966-frequency-of-the-most-frequent-element/frequency-of-the-most-frequent-element.js
var maxFrequency = function(nums, k) {
    nums.sort((a, b)=> a - b)
    let i = 0
    let j = 0
    let runningSum = 0
    let output = 0

    while (j < nums.length) {
        runningSum += nums[j]

        while (nums[j] * (j - i + 1) > runningSum + k) {
            runningSum -= nums[i]
            i += 1
        }

        output = Math.max(output, (j - i + 1))
        j += 1
    }

    return output

};

File: 1966-frequency-of-the-most-frequent-element/frequency-of-the-most-frequent-element.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()
        max_freq = 1
        curr_sum = 0
        i = 0
        j = 0

        while (j < len(nums)):
            curr_sum += nums[j]

            while (nums[j] * (j - i + 1) > curr_sum + k):
                curr_sum -= nums[i]
                i += 1

            max_freq = max(max_freq, j - i + 1)
            j += 1

        return max_freq
        
            

            